alcs-frontend/src/app/features/notice-of-intent/decision/decision-v2/decision-input/decision-input-v2.component.ts

```diff
@@ -0,0 +1,483 @@
+                this.existingDecision.uuid !== decision.uuid
+              ) {
+                this.isFirstDecision = false;
+              }
+            }
+
+            if (minDate) {
+              this.minDate = new Date(minDate);
+            }
+
+            if (!this.isFirstDecision) {
+              this.form.controls.postDecision.addValidators([Validators.required]);
+              this.form.controls.decisionMaker.disable();
+            }
+          } else {
+            this.isFirstDecision = true;
+          }
+        } else {
+          this.resolutionYearControl.enable();
+        }
+      });
+  }
+
+  private mapPostDecisionsToControls(
+    modifications: NoticeOfIntentModificationDto[],
+    existingDecision?: NoticeOfIntentDecisionDto
+  ) {
+    //TOOD: Clean this up
+    const mappedModifications = modifications
+      .filter(
+        (modification) =>
+          (existingDecision && existingDecision.modifies?.uuid === modification.uuid) ||
+          (modification.reviewOutcome.code === 'PRC' && !modification.resultingDecision)
+      )
+      .map((modification, index) => ({
+        label: `Modification Request #${modifications.length - index} - ${modification.modifiesDecisions
+          .map((dec) => `#${dec.resolutionNumber}/${dec.resolutionYear}`)
+          .join(', ')}`,
+        uuid: modification.uuid,
+      }));
+
+    this.postDecisions = [...mappedModifications];
+  }
+
+  private patchFormWithExistingData(existingDecision: NoticeOfIntentDecisionDto) {
+    this.form.patchValue({
+      outcome: existingDecision.outcome.code,
+      decisionMaker: existingDecision.decisionMaker,
+      date: existingDecision.date ? new Date(existingDecision.date) : undefined,
+      resolutionYear: existingDecision.resolutionYear,
+      resolutionNumber: existingDecision.resolutionNumber?.toString(10) || undefined,
+      auditDate: existingDecision.auditDate ? new Date(existingDecision.auditDate) : undefined,
+      postDecision: existingDecision.modifies?.uuid,
+      isSubjectToConditions: parseBooleanToString(existingDecision.isSubjectToConditions),
+      decisionDescription: existingDecision.decisionDescription,
+      isStatsRequired: parseBooleanToString(existingDecision.isStatsRequired),
+      rescindedDate: existingDecision.rescindedDate ? new Date(existingDecision.rescindedDate) : undefined,
+      rescindedComment: existingDecision.rescindedComment,
+    });
+
+    this.conditions = existingDecision.conditions;
+
+    if (existingDecision.isSubjectToConditions) {
+      this.showConditions = true;
+    }
+
+    if (!existingDecision.resolutionNumber) {
+      this.resolutionYearControl.enable();
+    }
+
+    if (existingDecision?.components) {
+      this.components = existingDecision.components;
+    }
+
+    this.requireComponents = ['APPR', 'APPA'].includes(existingDecision.outcome.code);
+
+    if (['APPR', 'APPA', 'RESC'].includes(existingDecision.outcome.code)) {
```

Comment:
nitpick: I noticed that there are a lot of places where this combination is used ['APPR', 'APPA', 'RESC']. Would be nice to move it to constant with nice name

Reply:
Added an enum in the main v2-component and imported for all NOI usage